common/lib/driver-definitions/src/urlResolver.ts

```diff
@@ -51,7 +51,7 @@ export interface DriverPreCheckInfo {
      * @deprecated - only needed as long as long as Loader.request() does not work as intended. When
      * Loader.request() caches and resolves pathing properly, this can be removed. #4489, #4491
      */
-    containerPath: string;
```

Comment:
I do not see a good reason for this interface to be here. Why not move it next to the only place it's used - checkUrl() and avoid all the compat changes you would need to go through every time you need to change it? I think this change would have fewer files in it if we started like that.

Reply:
> I do not see a good reason for this interface to be here. Why not move it next to the only place it's used - checkUrl() and avoid all the compat changes you would need to go through every time you need to change it? I think this change would have fewer files in it if we started like that.

I believe we did discuss moving this interface all together, but we thought it was out of the scope of this issue and should have its own.

> While not directly related to this change, I'm not sure I can understand the purpose of checkUrl().
It says that it can return false positives, and it's faster way to check URL.
But looking at usage of getLocatorFromOdspUrl(), it's clear that it's not required in OdspDriverUrlResolverForShareLink.resolve() to have "nav" parameter, nor it's tested in base resolver.
So it looks like it does return false negatives, no?

I'm not sure about this, @ChumpChief or @heliocliu , would either of you know?

Reply:
🤷‍♂️
Vlad raises a good point about the bigger picture though.  If this API's usage is so narrow and that usage is questionable, it makes sense to look into that first and do a "full" cleanup if applicable.

Reply:
Yeah, that definitely makes sense. We can move the interface definition at the same time that we change/remove the `containerPath` property.

Reply:
@vladsud is right. Both `OdspDriverUrlResolverForShareLink` and `OdspDriverUrlResolver` don't require `nav` search param. `checkUrl` returning `undefined` for a documentUrls that odsp drivers would process, is a false negative. We probably have a bug here.

This function seems to have some sort of identity crisis. Maybe initially it was intended as a boolean *check*, but then it turned into a data source. @scottn12 we could break this down into two parts. First you could check if the data this function provides is actually useful. I see Bohemia using it only at a few places. In `generateOdspDriverInfoFromUrl`  Im not sure why even use `checkUrl` if they are immediately calling `getLocatorFromOdspUrl` that would have provided them the same information. If not useful, lets aim to fully get rid of this call. Additionally, we could investigate if the boolean check is useful (valid/not valid), and then provide the desired API making sure we avoid false negatives.